nemo/collections/nlp/modules/common/megatron/bert.py

```diff
@@ -0,0 +1,139 @@
+from megatron.model.language_model import get_language_model
+from megatron.model.utils import init_method_normal, scaled_init_method_normal
+
+from nemo.collections.nlp.modules.common.bert_module import BertModule
+from nemo.core.classes import typecheck
+from nemo.utils.decorators import experimental
+
+__all__ = ['MegatronBertEncoder']
+
+
+@experimental
+class MegatronBertEncoder(BertModule):
+    """
+    MegatronBERT wraps around the Megatron Language model
+    from https://github.com/NVIDIA/Megatron-LM
+
+    Args:
+        config_file (str): path to model configuration file.
+        vocab_file (str): path to vocabulary file.
+        tokenizer_type (str): tokenizer type, currently only 'BertWordPieceLowerCase' supported.
+    """
+    def input_ports(self):
+        """Returns definitions of module input ports.
+        input_ids: input token ids
+        token_type_ids: segment type ids
+        attention_mask: attention mask
+        """
+        return {
+            "input_ids": NeuralType(('B', 'T'), ChannelType()),
+            "attention_mask": NeuralType(('B', 'T'), ChannelType()),
+            "token_type_ids": NeuralType(('B', 'T'), ChannelType(), optional=True),
+        }
+
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        hidden_states: output embedding 
+        """
+        return {"hidden_states": NeuralType(('B', 'T', 'D'), ChannelType())}
+
+    def __init__(
+        self,
+        model_name,
+        vocab_file,
+        hidden_size=1024,
+        num_attention_heads=16,
+        num_layers=24,
+        max_seq_length=512,
+        tokenizer_type='BertWordPieceLowerCase',
+        init_method_std=0.02,
+        num_tokentypes=2,
+    ):
+
+        super().__init__()
+        
+        if not os.path.exists(vocab_file):
+            raise ValueError(f'Vocab file not found at {vocab_file}')
+
+        megatron_args = {
+            "num_layers": num_layers,
+            "hidden_size": hidden_size,
+            "num_attention_heads": num_attention_heads,
+            "max_position_embeddings": max_seq_length,
+            "tokenizer_type": tokenizer_type,
+            "vocab_file": vocab_file,
+        }
+
+        initialize_megatron(None, megatron_args, ignore_unknown_args=True)
+        init_method = init_method_normal(init_method_std)
+
+        self.language_model, self._language_model_key = get_language_model(
+            attention_mask_func=bert_attention_mask_func,
+            num_tokentypes=num_tokentypes,
+            add_pooler=False,
+            init_method=init_method,
+            scaled_init_method=scaled_init_method_normal(init_method_std, num_layers),
+        )
+
+        self._hidden_size = self.language_model.hidden_size
+
+    @property
+    def hidden_size(self):
+        """
+            Property returning hidden size.
+
+            Returns:
+                Hidden size.
+        """
+        return self._hidden_size
```

Comment:
the rest of the Bert models inherit from HF models and use `self.hidden_size = self.bert_model.config.hidden_size` we need to make changes to Megatron to support this

Reply:
You mean, to megatron-lm ? Let me know if I can help.

Reply:
I ended up importing BertConfig from transformers and filling it out as other components expected it ...